GirafRest.Test/Controllers/DepartmentControllerTest.cs

```diff
@@ -80,26 +84,28 @@ public void Get_GetNonExistingDepartmentByID_NotFound()
 
             _testContext.MockUserManager.MockLoginAsUser(_testContext.MockUsers[UserGuardianDepartment1]);
 
-            var res = dc.Get(DEPARTMENT_ONE).Result;
+            var res = dc.Get(DEPARTMENT_ONE).Result as ObjectResult;
+
+            var body = res.Value as ErrorResponse;
 
-            Assert.False(res.Success);
-            Assert.Equal(ErrorCode.NotFound, res.ErrorCode);
+            Assert.Equal(StatusCodes.Status404NotFound, res.StatusCode);
+            Assert.Equal(ErrorCode.NotFound, body.ErrorCode);
         }
 
         [Fact]
         public void Get_AllDepartmentNames_Success()
         {
             var dc = initializeTest();
-            var res = dc.Get().Result;
+            var res = dc.Get().Result as ObjectResult;
+            var body = res.Value as SuccessResponse<List<DepartmentNameDTO>>;
 
-            Assert.True(res.Success);
-            Assert.Equal(res.ErrorCode, ErrorCode.NoError);
+            Assert.Equal(StatusCodes.Status200OK, res.StatusCode);
             // check data
-            Assert.Equal(_testContext.MockDepartments.Count(), res.Data.Count());
-            for (int i = 0; i < res.Data.Count; i++)
+            Assert.Equal(_testContext.MockDepartments.Count(), body.Data.Count());
+            for (int i = 0; i < body.Data.Count; i++)

```

Comment:
This could probably be changed to a LINQ statement for increased performance/compiler optimization.

Reply:
we can't directly compare them since they are lists of 2 different types